lib/bolt/plugin/module.rb

```diff
@@ -0,0 +1,240 @@
+# frozen_string_literal: true
+
+require 'bolt/task/run'
+
+module Bolt
+  class Plugin
+    class Module
+      class InvalidPluginData < Bolt::Plugin::PluginError
+        def initialize(plugin, msg)
+          msg = "Invalid Plugin Data for #{plugin}: #{msg}"
+          super(msg, 'bolt/invalid-plugin-data')
+        end
+      end
+
+      def self.load(name, modules, opts)
+        mod = modules[name]
+        if mod&.plugin?
+          opts[:mod] = mod
+          plugin = Bolt::Plugin::Module.new(opts)
+          plugin.setup
```

Comment:
In theory I can see the advantages but it means a bunch of class methods that have more arguments and don't think it actually allows us to do that much more real testing since nothing actually runs without a module on disk.